header-rewrite-filter/header_processor.cc

```diff
@@ -250,14 +259,72 @@ namespace HeaderRewriteFilter {
             return absl::InvalidArgumentError("error parsing boolean expression -- " + std::string(e.what()));
         }
 
+
         return absl::OkStatus();
     }
 
-    // return status and bool result
-    std::tuple<absl::Status, bool> SetBoolProcessor::executeOperation(bool negate) {
+    std::tuple<absl::Status, bool> SetBoolProcessor::executeOperation(Http::RequestOrResponseHeaderMap& headers, bool negate) {
+        bool result;
+        std::string source; // result of calling a function to fetch req/res metadata
+        switch (function_type_) {
+            case Utility::FunctionType::GetHdr:
```

Comment:
These are pretty big cases, and each function is a logical "thing" - maybe better to give them their own functions? e.g.
```
        switch (function_type_) {
            case Utility::FunctionType::GetHdr:
                dynf_gethdr(args)
            case Utility::FunctionType::Urlp:
                dynf_urlp(args)
```

That might also make it easier for people to add new functions in the future